x-pack/plugins/apm/server/routes/backends/get_top_backend_operations.ts

```diff
@@ -0,0 +1,192 @@
+  >;
+}
+
+export async function getTopBackendOperations({
+  setup,
+  backendName,
+  start,
+  end,
+  offset,
+  environment,
+  kuery,
+}: {
+  setup: Setup;
+  backendName: string;
+  start: number;
+  end: number;
+  offset?: string;
+  environment: Environment;
+  kuery: string;
+}) {
+  const { apmEventClient } = setup;
+
+  const { startWithOffset, endWithOffset, offsetInMs } = getOffsetInMs({
+    start,
+    end,
+    offset,
+  });
+
+  const aggs = {
+    duration: {
+      avg: {
+        field: SPAN_DURATION,
+      },
+    },
+    successful: {
+      filter: {
+        term: {
+          [EVENT_OUTCOME]: EventOutcome.success,
+        },
+      },
+    },
+    failure: {
+      filter: {
+        term: {
+          [EVENT_OUTCOME]: EventOutcome.failure,
+        },
+      },
+    },
+  };
+
+  const response = await apmEventClient.search('get_top_backend_operations', {
+    apm: {
+      events: [ProcessorEvent.span],
+    },
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            ...rangeQuery(startWithOffset, endWithOffset),
+            ...environmentQuery(environment),
+            ...kqlQuery(kuery),
+            ...termQuery(SPAN_DESTINATION_SERVICE_RESOURCE, backendName),
+          ],
+        },
+      },
+      aggs: {
+        operationName: {
+          terms: {
+            field: SPAN_NAME,
+            size: MAX_NUM_OPERATIONS,
+          },
+          aggs: {
+            over_time: {
+              date_histogram: getMetricsDateHistogramParams({
+                start: startWithOffset,
+                end: endWithOffset,
+                metricsInterval: 60,
+              }),
+              aggs,
+            },
+            ...aggs,
+            total_time: {
+              sum: {
+                field: SPAN_DURATION,
+              },
+            },
+          },
+        },
+      },
+    },
+  });
+
+  const getImpact = calculateImpactBuilder(
+    response.aggregations?.operationName.buckets.map(
+      (bucket) => bucket.total_time.value
+    ) ?? []
+  );
+
+  return (
+    response.aggregations?.operationName.buckets.map(
+      (bucket): BackendOperation => {
+        const timeseries: BackendOperation['timeseries'] = {
+          latency: [],
+          throughput: [],
+          failureRate: [],
+        };
+
+        bucket.over_time.buckets.forEach((dateBucket) => {
+          const x = dateBucket.key + offsetInMs;
+          timeseries.throughput.push({
+            x,
+            y: calculateThroughputWithInterval({
+              value: dateBucket.doc_count,
+              bucketSize: 60,
+            }),
+          });
+          timeseries.latency.push({ x, y: dateBucket.duration.value });
+          timeseries.failureRate.push({
+            x,
+            y:
+              dateBucket.failure.doc_count > 0 ||
+              dateBucket.successful.doc_count > 0
+                ? dateBucket.failure.doc_count /
```

Comment:
Can we have `dateBucket.failure.doc_count / dateBucket.doc_count` or we only take into account the events with a known outcome?
                   `

Reply:
the latter, `unknown` is ignored.